### ISSUE TITLE
Update some of our policies that are targeting `Pod`, to make them process higher level objects

### ISSUE THREAD

**Issue by jvanz** [kubewarden/kubewarden-controller#282]
Update some of our policies that are targeting `Pod`, to make them process higher level objects like deployments. This is a better practice because we prevent the resources to be created and then fail in the deployment phase when the pods are created.

**Comment by jvanz** [comment]
As we are not working on this now, I've re added the issue in the board to keep it in the "waiting triage" queue.

**Comment by flavio** [comment]
Instead of updating all our policies targeting Pods to cover for high order resources, we might consider doing something like [Kyverno aug-gen rules](https://main.kyverno.io/docs/writing-policies/autogen/)

**Comment by adnanhashmi09** [comment]
I would like to work on this issue. Please assign this to me.

**Comment by jvanz** [comment]
@adnanhashmi09, please, select a sub task. I'll assign it for you as well.

**Comment by flavio** [comment]
wait, I'm a bit confused. I don't know if @adnanhashmi09 wants to work on the original task or attempt to implement what I described inside of https://github.com/kubewarden/kubewarden-controller/issues/282#issuecomment-1665120267

@adnanhashmi09: can you please clarify?

**Comment by adnanhashmi09** [comment]
I am working on this [sub-issue.](https://github.com/kubewarden/volumes-psp-policy/issues/11). I left a comment there.

**Comment by flavio** [comment]
JFYI, I've created https://github.com/kubewarden/rfc/issues/23 to keep track of what I described inside of https://github.com/kubewarden/kubewarden-controller/issues/282#issuecomment-1665120267.

I suggest to put this issue on hold until we understand whether we can have a generic way to make Pod-specific policies work against higher-order resources

**Comment by niheetes** [comment]
Any progress on the generic way to make these policies work on higher level resources?

**Comment by jvanz** [comment]
> Any progress on the generic way to make these policies work on higher level resources?

No, this is issue is on hold for now. May I ask you why you're interested on this? Are you facing some issue writing policies or something similar? Do you want to contribute?

**Comment by jvanz** [comment]
@niheetes sorry for the long delay to give you a feedback. We've discussed this issue during our planning session and we do not want to fix all the policies manually. Because that will required a lot of changes and it not future proof. In other words, we would like to avoid to remember that policy must be handle pod inside high level resources every time. Therefore, we will refine this card to decide the proper way to move it forward. 

However, we do not want to leave you if a workaround. Thus, we would like to know which policies you want to get fix. Therefore, we can work on them while we do not find the final solution for this issue. Can you share that with us?

**Comment by jvanz** [comment]
I'm removing @adnanhashmi09 from assignee because there is no activity for a long time.
